analysis/dataset_definition_dx_to_med_time.py

```diff
@@ -0,0 +1,93 @@
+from ehrql import create_dataset
+from ehrql.tables.tpp import (
+    patients,
+    practice_registrations,
+    clinical_events,
+    medications,
+)
+
+from codelists import (
+    adhd_codelist,
+    adhdrem_codelist, 
+    methylphenidate_codelist
+)
+
+def last_matching_event(events, codelist, where=True):
+    """Select the last matching SNOMED CT event from specified codelist
+
+    Args:
+        events (e.g., clinical_events or ): Many rows per patient event frame to select last matching event from codelist from
+        codelist (codelist): Clinical codelist, must be using snomedct codes
+        where (bool, optional): _description_. Defaults to True.
+
+    Returns:
+        patient frame: One row per patient frame, with the last matching event from codelist
+    """
+    return (
+        events.where(where)
+        .where(events.snomedct_code.is_in(codelist))
+        .sort_by(events.date)
+        .last_for_patient()
+    )
+
+dataset = create_dataset()
+dataset.configure_dummy_data(population_size=5)
+
+# Date range
+start_date = "2016-04-01"
+end_date = "2017-03-31"
+
+# Population variables
+has_registration = practice_registrations.spanning(
+    start_date, end_date
+).exists_for_patient()
+dataset.sex = patients.sex
+dataset.age = patients.age_on(end_date)
+
+# Setting up the dates
+selected_events = clinical_events.where(
+    clinical_events.date.is_on_or_between(start_date, end_date)
+)
+
+selected_medications = medications.where(
+    medications.date.is_on_or_after(start_date)
+)
+
+# Filtering with the codelists
+has_adhd_event = selected_events.where(
+    clinical_events.snomedct_code.is_in(adhd_codelist)
+).exists_for_patient()
+
+# Picking the last date for dia and first date for med
+dataset.latests_adhd_diagnosis_date = last_matching_event(selected_events, adhd_codelist).date
+
+dataset.has_adhdrem_cod_date = last_matching_event(selected_events, adhdrem_codelist).date
```

Comment:
I think it'd be clearer if you followed the same naming approach you used for the variable above, e.g., here `latest_adhdrem_cod_date`

Reply:
Maybe we dont need this for this analysis? We want to count the time diff from dx to med for everyone regardless whether they have a remission code later?